lib/commons/dom/url-props-from-attribute.js

```diff
@@ -0,0 +1,154 @@
+/* global dom */
+
+/**
+ * Parse resource object for a given node from a specified attribute
+ * @method urlPropsFromAttribute
+ * @param {HTMLElement} node given node
+ * @param {String} attribute attribute of the node from which resource should be parsed
+ * @returns {Object}
+ */
+dom.urlPropsFromAttribute = function urlPropsFromAttribute(node, attribute) {
+	const value = node[attribute];
+	if (!value) {
+		return undefined;
+	}
+
+	const nodeName = node.nodeName.toUpperCase();
+	let parser = node;
+
+	/**
+	 * Note:
+	 * The need to create a parser, is to keep this function generic, to be able to parse resource from element like `iframe` with `src` attribute
+	 */
+	if (!['A', 'AREA'].includes(nodeName)) {
+		parser = document.createElement('a');
+		parser.href = value;
+	}
+
+	const { pathname = '', filename = '' } = getPathnameOrFilename(
+		parser.pathname
+	);
+	return {
+		protocol: parser.protocol,
+		hostname: parser.hostname,
+		port: getPort(parser.port),
+		pathname: pathname.replace(/\/$/, ''), // remove trialing slashes
```

Comment:
I'd recommend always adding the trailing slash, instead of removing it. It's not a path if it doesn't have the slash.